WIP refactor: using cobra and create subcommands

cpanato opened the pull request:
@tstromberg 

This is a refactor to make it a bit easier for future changes. If you like this I will continue on that and make it cleaner and remove some duplicate code.
Just in this way now to open this WIP PR to see what do you think.

But instead of having the `--kind` we split in subcommand and then we can have this

```
$ ./pullsheet
pullsheet - Generate spreadsheets based on GitHub contributions

pullsheet generates a CSV (comma separated values) & HTML output about GitHub activity across a series of repositories.

Usage:
  pullsheet [command]

Available Commands:
  help           Help about any command
  issue-comments Generate data around issues
  issues         Generate data around issues
  leaderboard    Generate leaderboard data
  prs            Generate data around pull requests
  reviews        Generate data around reviews

Flags:
  -h, --help                help for pullsheet
      --log-level string    the logging verbosity, either 'panic', 'fatal', 'error', 'warning', 'info', 'debug', 'trace' (default "info")
      --repos strings       comma-delimited list of repositories. ex: kubernetes/minikube, google/pullsheet
      --since string        when to query from
      --title string        Title to use for output pages
      --token-path string   GitHub token path
      --until string        when to query till
      --users strings       comma-delimiited list of users

Use "pullsheet [command] --help" for more information about a command.
```

and using cobra we have some handy helpers 😄 

please let me know your thoughts, if you disagree feel free to close it.
thanks!

tstromberg reviewed (approved):
Looks good to me, thank you!

cpanato commented:
Ok, I will continue on the refactor

cpanato commented:
@tstromberg the initial refactor is done, let me know what you think, we can improve more in other follow-ups 😄 

Added some missing license header and updated the readme

tstromberg commented:
Looks great. Thank you for organizing the code - it's more readable now.
